src/ast/rewriter/bv_rewriter.cpp

```diff
@@ -2803,6 +2803,27 @@ br_status bv_rewriter::mk_ite_core(expr * c, expr * t, expr * e, expr_ref & resu
     return BR_FAILED;
 }
 
+br_status bv_rewriter::mk_distinct(unsigned num_args, expr * const * args, expr_ref & result) {
+    if (num_args <= 1) {
+        result = m().mk_true();
+        return BR_DONE;
+    }
+    unsigned sz = get_bv_size(args[0]);
+    // check if num_args > 2^sz
```

Comment:
this may be clever code, but not easy to read.
Here is something simpler:

```
if (sz >= 32) 
    return BR_FAILED;
if (num_args <= 1u << sz)
    return BR_FAILED;
result = m().mk_false();
return BR_DONE;        
```